backend/src/oe/api/op-statistics.service.ts

```diff
@@ -1,7 +1,7 @@
-import {Promise} from "bluebird";
-import bitcoinApi from "../../api/bitcoin/bitcoin-api-factory";
-import logger from "../../logger";
-import {BlockHeight} from "./interfaces/op-energy.interface";
+import { Promise } from 'bluebird';
+import logger from '../../logger';
+import opBlockHeaderService from '../service/op-block-header.service';
+import { BlockHeader, BlockHeight } from './interfaces/op-energy.interface';
 
 export class OpStatisticService {

```

Comment:
As far as I can see: there are many algorithms for calculating variance (and std dev is sqrt(variance)), so I am not sure if this one is optimal, but I can propose to introduce a table nbdrStatisticCache( blockHeight, blockSpan, avg, stddev) to not to recalculate result multiple times

Reply:
Please ignore this as I and Thomas are improving this moving forward. We have some plans to optimize this.